### PR TITLE
Fields

### PR DESCRIPTION
**Pull request by willghatch**
I liked the name `Object.fields` more than `Object.keys`, because `Object.keys` exists in JS and returns a list of keys (an array of strings), not an object like we are returning from it.

### CODE REVIEW

File: docs/src/changelog/index.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Versions and changes-within-versions are listed in reverse-chronological order:
 
 + @{rcHead("0.1.11-rc7")}
 + 2022/08/03: Added `{!rsh} isDataVariant`.
++ 2022/08/03: Added `{!rsh} Struct.fields` and `{!rsh} Object.fields`.
```
Comment (jeapostrophe) [inline docs/src/changelog/index.md:12]: In the future, they should be in order of commits. No need to change this.

File: docs/src/rsh/compute/index.md [hunk 2]
```diff
@@ -1279,9 +1279,21 @@ these fields may be accompanied by additional fields specified afterwards.
 
 Shorthand for `{!rsh} { x: x, z: 5}`, where `{!rsh} x` is any bound identifier.
 
+#### `Object.fields`
+@{ref("rsh", "Object.fields")}
+
+```reach
+const o1 = Object({a: UInt, b: UInt});
+const fields = Object.fields(o1); // {a: UInt, b: UInt}
+const o2 = Object(...fields, c: UInt));
```
Comment (jeapostrophe) [inline docs/src/rsh/compute/index.md:1288]:
```suggestion
const o2 = Object({...fields, c: UInt}));
```

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
Nice, very small change

**Review by jeapostrophe — APPROVED** [review]

**Comment by jeapostrophe** [comment]
The keyword file might be wrong now

File: docs/src/changelog/index.md (hunk 1 above)
Comment (willghatch) [reply docs/src/changelog/index.md]: I also realize now that I didn't change the date after copy/pasting either.  So I'm fixing the order and the date.

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**